source/octf/trace/parser/ParsedIoTraceEventHandler.cpp

```diff
@@ -530,4 +532,174 @@ void ParsedIoTraceEventHandler::getFilePath(uint64_t devId,
+        std::string basename = "";
+
+        FileName key;
+        key.Id = id;
+        key.DeviceId = m_partId;
+
+        auto iter = m_fsNames.find(key);
+        if (iter != m_fsNames.end()) {
+            auto i = iter->Name.rfind('.', iter->Name.length());
+            if (i != std::string::npos) {
+                basename = iter->Name.substr(0, i);
+            }
+        }
+
+        return basename;
+    }
+
+    virtual std::string getFileName(uint64_t id) const override {
+        FileName key;
+        key.Id = id;
+        key.DeviceId = m_partId;
+
+        auto iter = m_fsNames.find(key);
+        if (iter != m_fsNames.end()) {
+            return iter->Name;
+        }
+
+        return "";
+    }
+
+    virtual std::string getFileExtension(uint64_t id) const override {
+        std::string extension = "";
+
+        FileName key;
+        key.Id = id;
+        key.DeviceId = m_partId;
+
+        auto iter = m_fsNames.find(key);
+        if (iter != m_fsNames.end()) {
+            auto length = iter->Name.length();
+            auto i = iter->Name.rfind('.', length);
+            if (i != std::string::npos) {
+                extension = iter->Name.substr(i + 1, length);
+            }
+        }
+
+        return extension;
+    }
+
+    virtual std::string getDirPath(uint64_t id) const override {
+        std::string dir = "";
+
+        FileName key;
+        key.Id = id;
+        key.DeviceId = m_partId;
+
+        auto iter = m_fsNames.find(key);
+        if (iter != m_fsNames.end()) {
+            getPath(iter->ParentId, dir);
+        }
+
+        return dir;
+    }
+
+    virtual std::string getFilePath(uint64_t id) const override {
+        std::string path = "";
+
+        FileName key;
+        key.Id = id;
+        key.DeviceId = m_partId;
+
+        auto iter = m_fsNames.find(key);
+        if (iter != m_fsNames.end()) {
+            path = getDirPath(id);
+
+            if (path != "/") {
+                path += "/";
+            }
+
+            path += iter->Name;
+        }
+
+        return path;
+    }
+
+    virtual uint64_t getParentId(uint64_t id) const override {
+        FileName key;
+        key.Id = id;
+        key.DeviceId = m_partId;
+
+        auto iter = m_fsNames.find(key);
+        if (iter != m_fsNames.end()) {
+            return iter->ParentId;
+        }
+
+        return 0;
+    }
+
+private:
+    void getPath(uint64_t id, std::string &path) const {
+        FileName key;
+        key.Id = id;
+        key.DeviceId = m_partId;
+
+        auto iter = m_fsNames.find(key);
+        if (iter != m_fsNames.end()) {
+            const auto &name = *iter;
+
+            if (name.Id != name.ParentId) {
+                getPath(name.ParentId, path);
+
+                if (!path.empty() && '/' != path.back()) {
+                    path += "/";
+                }
+            }
+
+            path += name.Name;
+        } else {
+            if (path != "") {
+                path += "../";
```

Comment:
Shouldn't we add the first slash like above?
  if (!path.empty() && '/' != path.back()) {
                    path += "/";